### DISCUSSION TITLE
Where do we find the dockerfiles for images used in the CI?

### DISCUSSION THREAD

**Discussion by GitMensch** [KDAB/hotspot#563] [Q&A] [answered]
After learning more about docker containers I _guess_ those mostly are based on common distro containers and add  the necessary tools on top. Are the dockerfiles for the images posted to ghrc.io public?
Side note: the public images can be accessed without a token also from the outside, for example to base own containers on them for gitpod; but after tinkering with that and seen how much work it is to add gitpod functions on these I think about doing it the other way around (copying the definitions from KDAB's dockerfiles to extend the one gitpod provides).

... and: would you consider to add some things to the KDAB docker files like perf (which then allows to work with the record files we have in the repo)?

This isn't cleaned up yet, but a working perf can be integrated with

```dockerfile
### PERF
RUN apt install -y bison flex libelf-dev libnewt-dev libdw-dev libaudit-dev libiberty-dev libunwind-dev \
  libcap-dev libzstd-dev liblzma-dev libnuma-dev libssl-dev systemtap-sdt-dev libbabeltrace-ctf-dev \
  libperl-dev libtraceevent-dev \
  binutils-dev gcc-multilib \
  python3-dev \
  libgtk2.0-dev
#  asciidoc                 optional, only for manpages, depdencies are huge

# TODO: leave clean "install only", in this case only the following dependencies are needed
# RUN apt install -y linux-tools-generic libtraceevent1 libpython3-dev

# see https://kernel.googlesource.com/pub/scm/linux/kernel/git/acme/linux/+refs
# ENV PERF_VER=perf-tools-fixes-for-v6.1-1-2023-01-06
ENV PERF_VER=cbe7d3e3c90b34dc3d12041911f3ecf3d28d7ad5
ENV KERNEL_REPO=https://github.com/torvalds/linux.git
ENV LINUX_ROOT=/opt/linux

RUN mkdir -p ${LINUX_ROOT}
WORKDIR ${LINUX_ROOT}/..
RUN git clone --depth 1 --filter=blob:none --sparse ${KERNEL_REPO}
WORKDIR ${LINUX_ROOT}
RUN git sparse-checkout set tools scripts arch
RUN git fetch --depth 1 origin ${PERF_VER}
RUN make -C tools/perf -j $(nproc) PYTHON=/usr/bin/python3

ENV PERF_EXEC_PATH=${LINUX_ROOT}/tools/perf
ENV PATH=${PERF_EXEC_PATH}:${PATH}

RUN sysctl kernel.perf_event_paranoid=-1
```

**Comment by milianw** [comment, marked as answer]
they are all part of the hotspot sources:

https://github.com/KDAB/hotspot/tree/master/scripts/appimage
https://github.com/KDAB/hotspot/tree/master/scripts/compile-test
